windows-agent/internal/proservices/landscape/service.go

```diff
@@ -81,9 +81,9 @@ func New(ctx context.Context, conf Config, db *database.DistroDB, args ...Option
 	}
 
 	if opts.homedir == "" {
-		homeDir := os.Getenv("UserProfile")
-		if homeDir == "" {
-			return nil, errors.New("could not locate the user profile dir: %UserProfile% is not set")
+		homeDir, err := os.UserHomeDir()
+		if homeDir == "" || err != nil {
+			return nil, errors.New("could not locate the user home dir")
```

Comment:
If `homeDir` would come empty an errror is returned instead, so checking the `err` variable is enough.